src/GNDStk/gnds-utility.hpp

```diff
@@ -0,0 +1,231 @@
+
+// -----------------------------------------------------------------------------
+// Macros, variables, enums
+// -----------------------------------------------------------------------------
+
+// printval
+// printvar
+// Are the same, so we don't need to remember the exact terminology
+#define printval(var) std::cout << #var ": [" << var << "]" << std::endl
+#define printvar(var) printval(var)
+
+// indent
+// Number of spaces of indentation you want, in the output of certain types
+inline int indent = 3;
+
+// format
+enum class format {
+   null, // default, automagick, etc.
+   tree, // our own simple text format
+   xml,
+   json,
+   hdf5
+};
+
+
+
+// -----------------------------------------------------------------------------
+// Forward declarations: general
+// -----------------------------------------------------------------------------
+
+// fixme Some reorganization elsewhere may make some of these unnecessary
```

Comment:
👍 I think some reorganization will greatly reduce the need for this. It would also make things more clear.

Reply:
At some point I wrote a long(er) comment about this and why it's in there. For the time being, it wasn't really viable to substantially reduce the number of forward declarations, without reducing clarity somewhere else. I think we both agree that one would ideally order things in such a way that few if any forward declarations are needed, and it's what I normally try to do.